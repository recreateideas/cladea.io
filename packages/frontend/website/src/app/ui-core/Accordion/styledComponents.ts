import styled, { css } from 'styled-components';
import {
    Accordion as MUIAccordion,
    AccordionDetails as MUIAccordionDetails,
    AccordionSummary as MUIAccordionSummary,
} from '@mui/material';

export const Container = styled.div``;
export const Accordion = styled(MUIAccordion)`
    ${(props) => {
        const {
            theme: {
                dsl: { palette },
            },
        } = props;
        return css`
            background-color: ${palette.neutrals.secondaryBg};
            box-shadow: none;
            background-image: none;
        `;
    }}
`;
export const AccordionDetails = styled(MUIAccordionDetails)`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    layout: { padding },
                },
            },
        } = props;
        return css`
            padding: 0px ${padding.small} ${padding.small} ${padding.small};
        `;
    }}
`;
export const AccordionSummary = styled(MUIAccordionSummary)`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    sizes,
                    layout: { padding },
                },
            },
        } = props;
        return css`
            &.Mui-expanded {
                min-height: ${sizes.accordion.medium};
            }
            > div {
                margin: ${padding.small} 0 !important;
            }
            padding: 0px ${padding.small};
        `;
    }}
`;
export const Title = styled.div``;
