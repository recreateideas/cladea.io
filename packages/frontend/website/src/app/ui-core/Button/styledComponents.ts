import styled, { css } from 'styled-components';
import { Button } from '@mui/material';

export const StyledButton = styled(Button)`
    ${(props) => {
        const {
            theme: {
                dsl: { palette },
            },
        } = props;
        return css`
            &.MuiButton-containedPrimary .MuiButton-label {
                color: ${palette.fonts.primaryInverted};
            }
        `;
    }}
`;
