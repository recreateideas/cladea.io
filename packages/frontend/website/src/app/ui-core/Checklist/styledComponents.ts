import styled from 'styled-components';

export const Container = styled.div`
    width: 100%;
    ul {
        background-color: inherit;
        .MuiListItemIcon-root {
            min-width: auto;
        }
        .MuiButtonBase-root {
        }
    }
`;
