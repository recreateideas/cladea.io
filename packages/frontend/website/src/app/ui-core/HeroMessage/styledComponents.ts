import styled, { css } from 'styled-components';

export const Container = styled.div<{ type: 'error' | 'warn' | 'success' }>`
    ${(props) => {
        const {
            theme: {
                dsl: { hexToRgba, styles, layout, palette },
            },
            type,
        } = props;
        const colorMap: {
            [type: string]: {
                color: string;
                backgroundColor: string;
            };
        } = {
            error: {
                color: palette.fonts.error,
                backgroundColor: hexToRgba(palette.secondary.orange[500], 10),
            },
            success: {
                color: palette.fonts.success,
                backgroundColor: hexToRgba(palette.secondary.green[500], 10),
            },
            warn: {
                color: palette.fonts.warn,
                backgroundColor: hexToRgba(palette.secondary.yellow[500], 10),
            },
        };
        return css`
            ${colorMap[type]}
            white-space: pre;
            padding: ${layout.padding.small};
            border-radius: ${styles.borderRadius};
            overflow-x: auto;
            .section {
                display: flex;
                .icon {
                    margin-right: ${layout.margin.small};
                }
            }
        `;
    }}
`;
