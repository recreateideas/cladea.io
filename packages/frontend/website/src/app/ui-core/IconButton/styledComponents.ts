import styled, { css } from 'styled-components';

export const Container = styled.div<{ shape?: 'square'; disabled?: boolean }>`
    ${(props) => {
        const {
            theme: {
                dsl: { palette, styles },
            },
            disabled,
            shape,
        } = props;
        const bordersRadius = shape === 'square' ? styles.borderRadius : undefined;
        return css`
            ${bordersRadius &&
            css`
                .MuiButtonBase-root {
                    border-radius: ${bordersRadius};
                }
            `}
            ${disabled &&
            css`
                &,
                * {
                    cursor: no-drop;
                    svg {
                        fill: ${palette.neutrals.inactive} !important;
                    }
                }
            `}
        `;
    }}
`;
