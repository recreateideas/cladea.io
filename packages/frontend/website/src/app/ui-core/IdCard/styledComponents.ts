import styled, { css } from "styled-components";
import { avatar } from "../images";

export const Container = styled.div<{ breakpoint: string }>`
  ${(props) => {
    const {
      theme: {
        dsl: { layout, palette },
      },
      breakpoint,
    } = props;
    const isSmall = breakpoint === "sm";
    const width = isSmall ? "100%" : "30vw";
    return css`
      border-radius: 12px;
      min-width: ${width};
      background-color: ${palette.neutrals.quaternaryBg};
      border: solid 1px ${palette.neutrals.borders};
      overflow: hidden;
      .background {
        width: 100%;
        height: 60vh;
        fill: ${palette.neutrals.secondaryBg};
      }
      z-index: ${layout.namedZIndex.leftSidebar - 1};
      .bio {
        display: flex;
        flex-direction: column;
      }
      .avatar {
        position: relative;
        border-radius: 100%;
        overflow: hidden;
        background: ${palette.neutrals.primaryBg};
        width: 20vh;
        height: 20vh;
        margin: 24px auto;
        .image-container {
          border-radius: 100%;
          overflow: hidden;
          width: inherit;
          height: inherit;
          background-image: url(${avatar});
          background-position: center;
          background-repeat: no-repeat;
          background-size: cover;
        }
        .space {
          background: unset !important;
        }
      }
      .name {
        display: flex;
        letter-spacing: 1px;
        justify-content: center;
        gap: 16px;
        .first {
        }
        .last {
          color: ${palette.tertiary.pink[500]};
        }
      }
      .position {
        width: 100%;
        .title {
          color: ${palette.secondary.neon[500]};
          text-align: center;
        }
      }
      .contacts {
        display: flex;
        flex-direction: column;
        gap: 16px;
        margin: ${isSmall ? "2vh" : "24px"} 24px;
        padding: ${isSmall ? "2vh" : "24px"};
        border: solid 1px ${palette.neutrals.borders};
        border-radius: 8px;
        background-color: ${palette.neutrals.secondaryBg};
        .contact-container {
          width: 100%;
          /* height: 40px; */
          display: flex;
          justify-content: center;
          .href-container {
            text-decoration: none;
            display: flex;
            gap: 16px;
            color: ${palette.fonts.primary};
            cursor: pointer;
            img {
              width: 36px;
            }
            .href-text {
              display: flex;
              flex-direction: column;
              justify-content: center;
              width: 80px;
            }
          }
        }
      }
    `;
  }};
`;

export const ToastContainer = styled.div`
  ${(props) => {
    const {
      theme: {
        dsl: { palette },
      },
    } = props;
    return css`
      .toast {
        display: block;
        position: fixed;
        top: 50%;
        left: 50%;
        transform: translateX(-50%);
        min-width: 450px;
        background-color: ${palette.neutrals.quaternaryBg};
        box-shadow: ${palette.neutrals.boxShadow(palette.secondary.neon[500])};
        border: solid 1px ${palette.neutrals.borders};
        .btn-close {
          color: ${palette.fonts.primary};
          background-color: ${palette.secondary.neon[500]};
        }
        .toast-header {
          justify-content: space-between;
          background-color: ${palette.neutrals.primaryBg};
          button {
            margin-right: 0;
          }
          .header-title {
            font-weight: 600;
            width: 100%;
            color: ${palette.secondary.neon[500]};
          }
        }

        .body-content {
          width: 100%;
          display: flex;
          justify-content: space-between;
          .copied-value {
            color: ${palette.tertiary.pink[500]};
          }
        }
      }
    `;
  }};
`;
