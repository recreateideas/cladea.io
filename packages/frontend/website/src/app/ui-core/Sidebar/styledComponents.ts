import styled, { css, FlattenInterpolation } from 'styled-components';

interface StylesMap {
    [type: string]: FlattenInterpolation<any>;
}

export const Container = styled.div<{ type: 'overlay'; from?: 'left' | 'right' }>`
    height: 100%;
    width: auto;
    display: flex;
    ${(props) => {
        const {
            type,
            from = 'left',
            theme: {
                dsl: {
                    layout: { namedZIndex },
                },
            },
        } = props;
        const flexDirection = from === 'left' ? 'row' : 'row-reverse';
        const styles: StylesMap = {
            overlay: css`
                position: absolute;
                z-index: ${namedZIndex.nodesSidebar};
                ${from}: 0px;
                flex-direction: ${flexDirection};
            `,
        };
        return css`
            ${styles[type]}
        `;
    }}
`;

export const Content = styled.div<{ isOpen: boolean; width?: number; from?: 'left' | 'right' }>`
    ${(props) => {
        const {
            isOpen,
            width = 200,
            from = 'left',
            theme: {
                dsl: {
                    palette: {
                        neutrals: { borders, secondaryBg, boxShadow },
                    },
                },
            },
        } = props;
        const borderSide = from === 'left' ? 'right' : 'left';
        const styles: StylesMap = {
            true: css`
                width: ${width}px;
                border-${borderSide}: solid 1px ${borders};
                box-shadow: ${boxShadow()};
            `,
            false: css`
                width: 0px;
                overflow: hidden;
            `,
        };
        return css`
            height: 100%;
            background-color: ${secondaryBg};
            transition: width 0.2s ease-in-out;
            ${styles[`${isOpen}`]}
        `;
    }}
`;

export const Handle = styled.div`
    position: relative;
    height: fit-content;
    border-radius: 100%;
    cursor: pointer;
    transition: all 0.2s ease-in-out;
    ${(props) => {
        const {
            theme: {
                dsl: { layout },
            },
        } = props;
        return css`
            top: ${layout.margin.xsmall};
        `;
    }}
`;
