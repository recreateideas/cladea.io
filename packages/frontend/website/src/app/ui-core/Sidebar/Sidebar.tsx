import { ReactNode, useState, ReactElement } from "react";
import { IconButton } from "src/app/ui-core";
import { ChevronRight, ChevronLeft } from "@mui/icons-material";
import { Container, Content, Handle } from "./styledComponents";

interface IProps {
  children: ReactNode;
  type?: "overlay";
  width?: number;
  from?: "left" | "right";
  openIcon?: ReactElement;
  closeIcon?: ReactElement;
}
export const Sidebar = (props: IProps): ReactElement => {
  const [isOpen, setIsOpen] = useState<boolean>(false);
  const {
    children,
    type = "overlay",
    width = 200,
    from = "left",
    openIcon = <ChevronLeft />,
    closeIcon = <ChevronRight />,
  } = props;
  const onToggle = () => setIsOpen((o) => !o);
  return (
    <Container className="sidebar" {...{ type, width, from }}>
      <Content className="sidebar-content" {...{ isOpen, width, from }}>
        <div>{children}</div>
      </Content>
      <Handle className="sidebar-handle" {...{ onClick: onToggle }}>
        <IconButton color="secondary">
          {isOpen ? closeIcon : openIcon}
        </IconButton>
      </Handle>
    </Container>
  );
};

Sidebar.displayName = "Sidebar";

Sidebar.defaultProps = {
  type: "overlay",
};
