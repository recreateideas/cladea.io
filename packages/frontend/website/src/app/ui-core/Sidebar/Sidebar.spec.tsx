import { Sidebar } from './Sidebar';
import { TestProvider } from 'src/tests';
import { ChevronRight, ChevronLeft, Menu, Add } from '@mui/icons-material';
import { Container, Content, Handle } from './styledComponents';

const { mountWithProviderAndTheme } = TestProvider();
describe('<Sidebar/>', () => {
    it('should render', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(
            <Sidebar>
                <div>some content</div>
            </Sidebar>,
            state
        );
        expect(root).toMatchSnapshot();
    });
    it('should render any children', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(
            <Sidebar>
                <div className="children">some content</div>
            </Sidebar>,
            state
        );
        const children = root.find('.children');
        expect(children.length).toBe(1);
    });
    it('should render the overlay type if passed as prop', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(
            <Sidebar type="overlay">
                <div className="children">some content</div>
            </Sidebar>,
            state
        );
        const container = root.find(Container);
        expect(container.props().type).toBe('overlay');
    });
    it('should open and close the sidebar when clicking on the handle', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(
            <Sidebar>
                <div className="children">some content</div>
            </Sidebar>,
            state
        );
        let content;
        let icon;
        const handle = root.find(Handle);
        handle.simulate('click');
        content = root.find(Content);
        // is open
        expect(content).toHaveStyleRule('width', '200px');
        icon = root.find(ChevronRight);
        expect(icon.length).toBe(1);
        handle.simulate('click');
        // is closed
        content = root.find(Content);
        icon = root.find(ChevronLeft);
        expect(content).toHaveStyleRule('width', '0px');
        expect(icon.length).toBe(1);
    });
    it('should use the icons passed as props', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(
            <Sidebar openIcon={<Menu />} closeIcon={<Add />}>
                <div className="children">some content</div>
            </Sidebar>,
            state
        );
        let content;
        let icon;
        const handle = root.find(Handle);
        handle.simulate('click');
        content = root.find(Content);
        // is open
        expect(content).toHaveStyleRule('width', '200px');
        icon = root.find(Add);
        expect(icon.length).toBe(1);
        handle.simulate('click');
        // is closed
        content = root.find(Content);
        icon = root.find(Menu);
        expect(content).toHaveStyleRule('width', '0px');
        expect(icon.length).toBe(1);
    });
    it('should align the sidebar to the right if from=right', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(
            <Sidebar from="right">
                <div className="children">some content</div>
            </Sidebar>,
            state
        );
        let content;
        const handle = root.find(Handle);
        handle.simulate('click');
        content = root.find(Container);
        expect(content).toHaveStyleRule('right', '0px');
    });
    it('should align the sidebar to the left if from=left', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(
            <Sidebar from="left">
                <div className="children">some content</div>
            </Sidebar>,
            state
        );
        let content;
        const handle = root.find(Handle);
        handle.simulate('click');
        content = root.find(Container);
        expect(content).toHaveStyleRule('left', '0px');
    });
    it('should use the width passed as props', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(
            <Sidebar width={111}>
                <div className="children">some content</div>
            </Sidebar>,
            state
        );
        let content;
        const handle = root.find(Handle);
        handle.simulate('click');
        content = root.find(Content);
        // is open
        expect(content).toHaveStyleRule('width', '111px');
    });
});
