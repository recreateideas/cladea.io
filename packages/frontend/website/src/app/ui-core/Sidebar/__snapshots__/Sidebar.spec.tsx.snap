// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar/> should render 1`] = `
Array [
  .c5 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
}

.c6 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c3::-moz-focus-inner {
  border-style: none;
}

.c3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c4 {
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0,0,0,0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  color: #12D8F7;
}

.c4:hover {
  background-color: rgba(0,0,0,0.04);
}

.c4:hover {
  background-color: rgba(18,216,247,0.04);
}

.c4.Mui-disabled {
  background-color: transparent;
  color: rgba(0,0,0,0.26);
}

.c0 {
  height: 100%;
  width: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  z-index: 50;
  left: 0px;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  height: 100%;
  background-color: #f9fafb;
  -webkit-transition: width 0.2s ease-in-out;
  transition: width 0.2s ease-in-out;
  width: 0px;
  overflow: hidden;
}

.c2 {
  position: relative;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  border-radius: 100%;
  cursor: pointer;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  top: 4px;
}

@media print {
  .c3 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c4:hover {
    background-color: transparent;
  }
}

@media (hover:none) {
  .c4:hover {
    background-color: transparent;
  }
}

<div
    className="c0 sidebar"
    from="left"
    type="overlay"
    width={200}
  >
    <div
      className="c1 sidebar-content"
      from="left"
      width={200}
    >
      <div>
        <div>
          some content
        </div>
      </div>
    </div>
    <div
      className="c2 sidebar-handle"
      onClick={[Function]}
    >
      <div
        className="icon-button"
      >
        <button
          className="c3 MuiButtonBase-root c4 MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeMedium"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="c5 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
            data-testid="ChevronLeftIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
          <span
            className="c6 MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "",
]
`;
